src/app/firebase_app.ts

```diff
@@ -516,6 +526,12 @@ export function createFirebaseNamespace(): FirebaseNamespace {
     // Monkey-patch the serviceNamespace onto the firebase namespace
     (namespace as any)[name] = serviceNamespace;
 
+    // Patch the FirebaseAppImpl prototype
+    FirebaseAppImpl.prototype[name] = function(...args) {
+      const serviceFxn = this._getService.bind(this, name);
+      return serviceFxn.apply(this, allowMultipleInstances ? args : []);
+    }
```

Comment:
It's late, maybe I'm misreading:
So this will patch in the service factory: app.storage(instanceString);
Will this pass instanceString to createService(app, extendApp, instanceString) ?

This statement, as quoted from above, doesn't pass the instance string.
let service = this.firebase_.INTERNAL.factories[name](this, this.extendApp.bind(this));

If you update the test below where I commented to confirm this, then you are good to go.